src/nodes/chunk_append/exec.c

```diff
@@ -32,10 +32,17 @@
 #define INVALID_SUBPLAN_INDEX (-1)
 #define NO_MATCHING_SUBPLANS (-2)
 
+typedef enum SubplanState
+{
+	SUBPLAN_STATE_INCLUDED = 1 << 0, /* Used and not removed by startup exclusion */
+	SUBPLAN_STATE_FINISHED = 1 << 1, /* The subplan is finished */
```

Comment:
Can it be not included and finished at the same time? Flags often imply that any combination of them is possible, might be better to use mutually exclusive states here.

Reply:
@akuzm `subplan_state` is accessed by two different indexes (position in `initial_subplans` and position in `filtered_subplans`). In an ideal world, we would have two different boolean arrays to store this information. However, we can have only one `FLEXIBLE_ARRAY_MEMBER` in our shared memory struct and I wanted to prevent a mapping between both indexes. I added a comment to explain this implementation aspect.

However, I have no strong opinion on this solution. I could also implement the mapping between both indexes and use mutually exclusive states. This would make the states more clear but requires some additional code.